File: leetcode/AIcoding/5.Qwen-3/java_solutions/1422-maximum-score-after-splitting-a-string.java
public class Solution {
    public int maxScore(String s) {
        int totalOnes = 0;
        for (char c : s.toCharArray()) {
            if (c == '1') {
                totalOnes++;
            }
        }

        int leftOnes = 0;
        int maxScore = 0;

        for (int i = 0; i < s.length() - 1; i++) {
            if (s.charAt(i) == '1') {
                leftOnes++;
            }
            int rightOnes = totalOnes - leftOnes;
            int score = leftOnes + rightOnes;
            if (score > maxScore) {
                maxScore = score;
            }
        }

        return maxScore;
    }
}

File: leetcode/AIcoding/2.Gemini-2.0/c_solutions/1422-maximum-score-after-splitting-a-string.c
#include <stdio.h>
#include <stdlib.h>
#include <string.h>

int maxScore(char * s){
    int len = 0;
    while(s[len] != '\0'){
        len++;
    }
    int max_score = 0;
    for(int i = 1; i < len; i++){
        int score = 0;
        for(int j = 0; j < i; j++){
            if(s[j] == '0'){
                score++;
            }
        }
        for(int j = i; j < len; j++){
            if(s[j] == '1'){
                score++;
            }
        }
        if(score > max_score){
            max_score = score;
        }
    }
    return max_score;
}